test/unittest/Database/DatabaseStatusTests.cpp

```diff
@@ -0,0 +1,729 @@
+    ASSERT_TRUE(datawriter->active);
+    ASSERT_TRUE(datareader->active);
+    ASSERT_TRUE(locator->active);
+    ASSERT_TRUE(datareader1->active);
+    ASSERT_TRUE(reader_locator1->active);
+}
+
+TEST_F(database_status_tests, endpoints)
+{
+    db.change_entity_status(datawriter->id, false);
+    db.change_entity_status(datareader->id, false);
+
+    ASSERT_TRUE(host->active);
+    ASSERT_TRUE(user->active);
+    ASSERT_TRUE(process->active);
+    ASSERT_TRUE(domain->active);
+    ASSERT_FALSE(topic->active);
+    ASSERT_TRUE(participant->active);
+    ASSERT_FALSE(datawriter->active);
+    ASSERT_FALSE(datareader->active);
+    ASSERT_TRUE(locator->active);
+
+    db.change_entity_status(datawriter->id, true);
+    db.change_entity_status(datareader->id, true);
+
+    ASSERT_TRUE(host->active);
+    ASSERT_TRUE(user->active);
+    ASSERT_TRUE(process->active);
+    ASSERT_TRUE(domain->active);
+    ASSERT_TRUE(topic->active);
+    ASSERT_TRUE(participant->active);
+    ASSERT_TRUE(datawriter->active);
+    ASSERT_TRUE(datareader->active);
+    ASSERT_TRUE(locator->active);
+
+    auto datawriter1 = std::make_shared<DataWriter>("datawriter1", "qos", "21.22.23.24", participant, topic);
+    auto writer_locator1 = std::make_shared<Locator>("writer_locator1");
+    writer_locator1->id = db.generate_entity_id();
+    datawriter1->locators[writer_locator1->id] = writer_locator1;
+    db.insert(datawriter1);
+
+    auto datareader1 = std::make_shared<DataReader>("datareader1", "qos", "11.12.13.14", participant, topic);
+    auto reader_locator1 = std::make_shared<Locator>("reader_locator1");
+    reader_locator1->id = db.generate_entity_id();
+    datareader1->locators[reader_locator1->id] = reader_locator1;
+    db.insert(datareader1);
+
+    db.change_entity_status(datawriter->id, false);
+    db.change_entity_status(datareader->id, false);
+
+    ASSERT_TRUE(host->active);
+    ASSERT_TRUE(user->active);
+    ASSERT_TRUE(process->active);
+    ASSERT_TRUE(domain->active);
+    ASSERT_TRUE(topic->active);
+    ASSERT_TRUE(participant->active);
+    ASSERT_FALSE(datawriter->active);
+    ASSERT_FALSE(datareader->active);
+    ASSERT_TRUE(locator->active);
+    ASSERT_TRUE(datareader1->active);
+    ASSERT_TRUE(reader_locator1->active);
+    ASSERT_TRUE(datawriter1->active);
+    ASSERT_TRUE(writer_locator1->active);
+
+    db.change_entity_status(datawriter->id, true);
+    db.change_entity_status(datareader->id, true);
+
+    ASSERT_TRUE(host->active);
+    ASSERT_TRUE(user->active);
+    ASSERT_TRUE(process->active);
+    ASSERT_TRUE(domain->active);
+    ASSERT_TRUE(topic->active);
+    ASSERT_TRUE(participant->active);
+    ASSERT_TRUE(datawriter->active);
+    ASSERT_TRUE(datareader->active);
+    ASSERT_TRUE(locator->active);
+    ASSERT_TRUE(datareader1->active);
+    ASSERT_TRUE(reader_locator1->active);
+    ASSERT_TRUE(datawriter1->active);
+    ASSERT_TRUE(writer_locator1->active);
+}
+
+TEST_F(database_status_tests, dds_entities)
```

Comment:
I believe this cases are already covered with the endpoint and participant cases in isolation. Could we remove this testcase?

Reply:
The aim of this test was to check that if we deactivate the participant and the endpoints all the database (except locators and domains) will be inactive. But maybe is not necessary, confirm me and I will remove it.

Reply:
The deactivation of endpoints and participants are kind of independent, in the sense that the effects of ones (deactivating the topic) is not affected by the effects on the other (deactivating the process-use-host). That's why I see this as redundant.

BUT

it does not make any harm, so if you think it makes sense to check this specifically, I will not oppose.

Reply:
You convince me, the deactivation of endpoints and participants are independent, this is completely redundant. I remove it, do not make sense for me keep a redundant test.